js/components/graph/Graph.js

```diff
@@ -338,6 +341,77 @@ export default class Graph extends React.Component {
     }
   };
 
+  /**
+   * Returns the position of the mouse adjusted to the viewbox's new origin. This is so that the change
+   * in viewboxX and viewboxY are relative to the shifted viewbox origin after panning.
+   * @param {double} clientX
+   * @param {double} clientY
+   * @return {Object}
+   */
+  adjustCoordsToViewbox = (clientX, clientY) => {
+    return {
+      adjustedX: clientX + this.state.viewBoxPos.x,
+      adjustedY: clientY + this.state.viewBoxPos.y
+    };
+  }
+
+  /**
+   * Initializes the panning process by recording the position of the mouse pointer and adding event listeners.
+   * @param {*} event
+   */
+  onMouseDownInGraph = event => {
+    document.body.style.cursor = "grab";
+    const {adjustedX, adjustedY} = this.adjustCoordsToViewbox(event.clientX, event.clientY);
+
+    document.getElementById("react-graph").addEventListener("mouseup", this.onMouseUpInGraph);
```

Comment:
It's not a good practice to add/remove event listeners in React; they should always be attached.

I think you mentioned encountering issues about two different event handlers being called; you can experiment with [`Event.preventDefault`](https://developer.mozilla.org/en-US/docs/Web/API/Event/preventDefault) or perhaps `Event.stopPropagation` to deal with this.